src/jit/liveness.cpp

```diff
@@ -1070,181 +1070,202 @@ VARSET_VALRET_TP Compiler::fgGetHandlerLiveVars(BasicBlock* block)
     return liveVars;
 }
 
-/*****************************************************************************
- *
- *  This is the classic algorithm for Live Variable Analysis.
- *  If updateInternalOnly==true, only update BBF_INTERNAL blocks.
- */
-
-void Compiler::fgLiveVarAnalysis(bool updateInternalOnly)
+class LiveVarAnalysis
 {
-    BasicBlock* block;
-    bool        change;
-#ifdef DEBUG
-    VARSET_TP VARSET_INIT_NOCOPY(extraLiveOutFromFinally, VarSetOps::MakeEmpty(this));
-#endif // DEBUG
-    bool keepAliveThis = lvaKeepAliveAndReportThis() && lvaTable[info.compThisArg].lvTracked;
+    Compiler* m_compiler;
 
-    /* Live Variable Analysis - Backward dataflow */
+    bool      m_changed;
+    bool      m_hasPossibleBackEdge;
 
-    bool hasPossibleBackEdge = false;
+    bool      m_heapLiveIn;
+    bool      m_heapLiveOut;
+    VARSET_TP m_liveIn;
+    VARSET_TP m_liveOut;
 
-    do
+    LiveVarAnalysis(Compiler* compiler)
+        : m_compiler(compiler), m_changed(false), m_hasPossibleBackEdge(false), m_heapLiveIn(false), m_heapLiveOut(false), m_liveIn(VarSetOps::MakeEmpty(compiler)), m_liveOut(VarSetOps::MakeEmpty(compiler))
     {
-        change = false;
-
-        /* Visit all blocks and compute new data flow values */
-
-        VARSET_TP VARSET_INIT_NOCOPY(liveIn, VarSetOps::MakeEmpty(this));
-        VARSET_TP VARSET_INIT_NOCOPY(liveOut, VarSetOps::MakeEmpty(this));
-
-        bool heapLiveIn  = false;
-        bool heapLiveOut = false;
+    }
 
-        for (block = fgLastBB; block; block = block->bbPrev)
+    void PerBlock(BasicBlock* block, bool updateInternalOnly, bool keepAliveThis)
```

Comment:
I would prefer a more meaningful name for this. `PerBlock()` could be something that just returns the liveness info for the block, rather than computing it. How about `PerBlockAnalysis`?